### PR TITLE
Add Models and PayloadListener

### PR DESCRIPTION
**Pull request by jackschofield23**
Signed-off-by: jackschofield23 <jack.schofield@answerdigital.com>

### CODE REVIEW

**Review by DanielNuttall-AD — APPROVED** [review]

File: src/PayloadListener/Extensions/ValidationExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
+﻿using Ardalis.GuardClauses;
```
Comment (mocsharp) [inline src/PayloadListener/Extensions/ValidationExtensions.cs:1]: missing copyright header

File: src/PayloadListener/Services/EventPayloadRecieverService.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
+﻿using Monai.Deploy.Messaging;
```
Comment (mocsharp) [inline src/PayloadListener/Services/EventPayloadRecieverService.cs:1]: missing copyright header

File: src/PayloadListener/Services/EventPayloadRecieverService.cs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+namespace Monai.Deploy.WorkflowManager.PayloadListener.Services
+{
+    public class EventPayloadRecieverService : IEventPayloadRecieverService
+    {
+        public EventPayloadRecieverService(
+            ILogger<EventPayloadRecieverService> logger,
+            IEventPayloadValidator payloadValidator,
+            IMessageBrokerSubscriberService messageBrokerSubscriberService)
+        {
+            Logger = logger;
```
Comment (mocsharp) [inline src/PayloadListener/Services/EventPayloadRecieverService.cs:17]: Should check for null

File: src/PayloadListener/Services/IEventPayloadRecieverService.cs [hunk 4]
```diff
@@ -0,0 +1,14 @@
+﻿using Monai.Deploy.Messaging.Common;
```
Comment (mocsharp) [inline src/PayloadListener/Services/IEventPayloadRecieverService.cs:1]: missing copyright header

File: src/PayloadListener/Validators/EventPayloadValidator.cs [hunk 5]
```diff
@@ -0,0 +1,49 @@
+﻿using Ardalis.GuardClauses;
```
Comment (mocsharp) [inline src/PayloadListener/Validators/EventPayloadValidator.cs:1]: missing copyright header

File: src/WorkflowManager/Logging/FileLoggingTextFormatter.cs [hunk 6]
```diff
@@ -61,13 +32,49 @@ protected override void AppendMessage(StringBuilder sb, string message)
...
 
             if (exception != null)
+            {
                 AppendException(sb, exception);
+            }
+        }
+
+        protected override void AppendTimestamp(StringBuilder sb, DateTimeOffset timestamp)
+        {
+            sb.Append(timestamp.ToLocalTime().ToString("o", CultureInfo.InvariantCulture)).Append(' ');
```
Comment (mocsharp) [inline src/WorkflowManager/Logging/FileLoggingTextFormatter.cs:52]: Should check for null params

File: src/WorkflowManager/Logging/FileLoggingTextFormatter.cs [hunk 7]
```diff
@@ -61,13 +32,49 @@ protected override void AppendMessage(StringBuilder sb, string message)
...
+        }
+
+        protected override void AppendLogScopeInfo(StringBuilder sb, IExternalScopeProvider scopeProvider)
+        {
+            scopeProvider.ForEachScope(
```
Comment (mocsharp) [inline src/WorkflowManager/Logging/FileLoggingTextFormatter.cs:57]: should check for null params

File: src/WorkflowManager/Logging/FileLoggingTextFormatter.cs [hunk 8]
```diff
@@ -61,13 +32,49 @@ protected override void AppendMessage(StringBuilder sb, string message)
...
+            {
+                builder.Append(' ');
+
+                AppendLogScope(builder, scope);
+            }, sb);
+        }
+
+        protected override void AppendLogScope(StringBuilder sb, object scope)
+        {
+            sb.Append('[').Append(scope).Append(']');
```
Comment (mocsharp) [inline src/WorkflowManager/Logging/FileLoggingTextFormatter.cs:68]: should check for null params

File: src/WorkflowManager/Logging/FileLoggingTextFormatter.cs [hunk 9]
```diff
@@ -61,13 +32,49 @@ protected override void AppendMessage(StringBuilder sb, string message)
...
+        }
+
+        protected override void AppendMessage(StringBuilder sb, string message)
+        {
+            sb.Append(" => ");
```
Comment (mocsharp) [inline src/WorkflowManager/Logging/FileLoggingTextFormatter.cs:73]: should check for null params.

File: tests/PayloadListener.Tests/Services/EventPayloadRecieverServiceTests.cs [hunk 10]
```diff
@@ -0,0 +1,83 @@
+﻿using NUnit.Framework;
```
Comment (mocsharp) [inline tests/PayloadListener.Tests/Services/EventPayloadRecieverServiceTests.cs:1]: missing copyright header

Comment (mocsharp) [reply tests/PayloadListener.Tests/Services/EventPayloadRecieverServiceTests.cs:1]: May also want to sort the usings; There's a Visual Studio extension `CodeMaid` for that.

File: tests/PayloadListener.Tests/Validators/EventPayloadValidatorTests.cs [hunk 11]
```diff
@@ -0,0 +1,150 @@
+﻿using System;
```
Comment (mocsharp) [inline tests/PayloadListener.Tests/Validators/EventPayloadValidatorTests.cs:1]: missing copyright header

File: tests/PayloadListener.Tests/Validators/EventPayloadValidatorTests.cs [hunk 12]
```diff
@@ -0,0 +1,150 @@
...
+using System.Collections.Generic;
+using Microsoft.Extensions.Logging;
+using Moq;
+using Monai.Deploy.WorkflowManager.PayloadListener.Validators;
+using NUnit.Framework;
```
Comment (mocsharp) [inline tests/PayloadListener.Tests/Validators/EventPayloadValidatorTests.cs:6]: I see we are using NUnit here while some other projects use xunit; do we want to discuss which one to use across all projects?

**Pull request closed without merging**